luajson.c

```diff
@@ -475,6 +475,10 @@ encode(lua_State *L, luaL_Buffer *b, int arg)
 				lua_pop(L, 1);
 				continue;
 			}
+			if (lua_type(L, arg + 1) != LUA_TSTRING) {
+				luaL_error(L, "Lua type %s is unsupported as object key",
+				    luaL_typename(L, arg));
```

Comment:
This should match up with the `arg + 1` surely?

But I think this is wrong anyway, since for these `arg`-relative references to be correct they need to be after the `lua_insert` calls below. At this point you have the latest key at `-2` and the latest value at `-1`, so you want `-2` in both the check and here.

Reply:
Should be fixed now, and added a test.